l10n_es_aeat_mod347/models/mod347.py

```diff
@@ -0,0 +1,737 @@
+                self._create_partner_records("A", KEY_TAX_MAPPING["A"])
+                self._create_partner_records("B", KEY_TAX_MAPPING["B"])
+                self._create_cash_moves()
+            self.recompute()
+            report.partner_record_ids.calculate_quarter_totals()
+        return True
+
+
+class L10nEsAeatMod347PartnerRecord(models.Model):
+    _name = "l10n.es.aeat.mod347.partner_record"
+    _inherit = ["mail.thread", "mail.activity.mixin", "portal.mixin"]
+    _description = "Partner Record"
+    _rec_name = "partner_vat"
+
+    @api.model
+    def _default_record_id(self):
+        return self.env.context.get("report_id", False)
+
+    report_id = fields.Many2one(
+        comodel_name="l10n.es.aeat.mod347.report",
+        string="AEAT 347 Report",
+        ondelete="cascade",
+        default=_default_record_id,
+    )
+    user_id = fields.Many2one(
+        comodel_name="res.users",
+        string="Salesperson",
+        track_visibility="onchange",
+        default=lambda self: self.env.user,
+        copy=False,
+    )
+    state = fields.Selection(
+        selection=[
+            ("pending", "Pending"),
+            ("sent", "Sent"),
+            ("confirmed", "Confirmed"),
+            ("exception", "Exception"),
+        ],
+        default="pending",
+        string="State",
+    )
+    operation_key = fields.Selection(
+        selection=[
+            (
+                "A",
+                u"A - Adquisiciones de bienes y servicios superiores al " u"límite (1)",
+            ),
+            ("B", u"B - Entregas de bienes y servicios superiores al límite (1)"),
+            ("C", u"C - Cobros por cuenta de terceros superiores al límite (3)"),
+            (
+                "D",
+                u"D - Adquisiciones efectuadas por Entidades Públicas "
+                u"(...) superiores al límite (1)",
+            ),
+            (
+                "E",
+                u"E - Subvenciones, auxilios y ayudas satisfechas por Ad. "
+                u"Públicas superiores al límite (1)",
+            ),
+            ("F", u"F - Ventas agencia viaje"),
+            ("G", u"G - Compras agencia viaje"),
+        ],
+        string="Operation Key",
+    )
+    partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Partner", required=True
+    )
+    partner_vat = fields.Char(string="VAT number", size=9)
+    representative_vat = fields.Char(
+        string="L.R. VAT number", size=9, help="Legal Representative VAT number"
+    )
+    community_vat = fields.Char(
+        string="Community vat number",
+        size=17,
+        help="VAT number for professionals established in other state "
+        "member without national VAT",
+    )
+    partner_country_code = fields.Char(string="Country Code", size=2)
+    partner_state_code = fields.Char(string="State Code", size=2)
+    first_quarter = fields.Float(
+        string="First quarter operations",
+        digits=dp.get_precision("Account"),
```

Comment:
```suggestion
        digits="Account",
```
Change in all fields to avoid warning:
https://github.com/odoo/odoo/blob/544116c55ea8938a4d85ddc200ed57427d5c3d43/odoo/addons/base/models/decimal_precision.py#L13